api/v1/etcd_types.go

```diff
@@ -17,135 +17,115 @@ limitations under the License.
-	Labels map[string]string `json:"labels,omitempty"`
-	// +optional
-	Annotations map[string]string `json:"annotations,omitempty"`
-	// +required
-	StorageClass string `json:"storageClass"`
+	StorageClass *string `json:"storageClass,omitempty"`
 	// +optional
-	TLSServerSecretName string `json:"tlsServerSecret,omitempty"`
-	// +optional
-	TLSClientSecretName string `json:"tlsClientSecret,omitempty"`
-	// +optional
-	StorageCapacity string `json:"storageCapacity,omitempty"`
+	StorageCapacity *string `json:"storageCapacity,omitempty"`
 }
 
 // StoreSpec defines parameters related to ObjectStore persisting backups
 type StoreSpec struct {
 	// +required
-	StorageContainer string `json:"storageContainer"`
+	Container string `json:"storageContainer"`
 	// +optional
-	StorePrefix string `json:"storePrefix,omitempty"`
+	Prefix *string `json:"storePrefix,omitempty"`
 	// +required
-	StorageProvider string `json:"storageProvider"`
+	Provider StorageProvider `json:"provider"`
 	// +required
-	StoreSecret string `json:"storeSecret"`
+	secretRef corev1.SecretReference `json:"secretRef"`
 }
 
-// BackupSpec defines parametes associated with the full and delta snapshots of etcd
-type BackupSpec struct {
-	// +optional
-	Port int `json:"port,omitempty"`
+// TLSConfig hold the TLS configuration detials.
+type TLSConfig struct {
 	// +required
-	ImageRepository string `json:"imageRepository"`
+	ServerTLSSecretRef corev1.SecretReference
 	// +required
-	ImageVersion string `json:"imageVersion"`
-	// +optional
-	FullSnapshotSchedule string `json:"fullSnapshotSchedule,omitempty"`
+	ClientTLSSecretRef corev1.SecretReference
+}
+
+// BackupSpec defines parametes associated with the full and delta snapshots of etcd
+type BackupSpec struct {
 	// +optional
-	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	Port *int `json:"port,omitempty"`
 	// +optional
-	PullPolicy corev1.PullPolicy `json:"pullPolicy,omitempty"`
+	TLS *TLSConfig `json:"tls,omitempty"`
+	// +required
+	Version string `json:"version"`
 	// +optional
-	GarbageCollectionPolicy GarbageCollectionPolicy `json:"garbageCollectionPolicy,omitempty"`
+	Store *StoreSpec `json:"store,omitempty"`
 	// +optional
-	GarbageCollectionPeriod string `json:"garbageCollectionPeriod,omitempty"`
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
 	// +optional
-	EtcdQuotaBytes int `json:"etcdQuotaBytes,omitempty"`
+	FullSnapshotSchedule *string `json:"fullSnapshotSchedule,omitempty"`
 	// +optional
-	EtcdConnectionTimeout string `json:"etcdConnectionTimeout,omitempty"`
+	GarbageCollectionPolicy *GarbageCollectionPolicy `json:"garbageCollectionPolicy,omitempty"`
 	// +optional
-	SnapstoreTempDir string `json:"snapstoreTempDir,omitempty"`
+	GarbageCollectionPeriod *time.Duration `json:"garbageCollectionPeriod,omitempty"`
 	// +optional
-	DeltaSnapshotPeriod string `json:"deltaSnapshotPeriod,omitempty"`
+	DeltaSnapshotPeriod *time.Duration `json:"deltaSnapshotPeriod,omitempty"`
```

Comment:
Shouldn't this be [`*metav1.Duration`](https://github.com/gardener/https://github.com/gardener/gardener/blob/master//vendor/k8s.io/apimachinery/pkg/apis/meta/v1/duration.go#L295) or [`*GardenerDuration`](https://github.com/gardener/https://github.com/gardener/gardener/blob/master//vendor/k8s.io/apimachinery/pkg/apis/meta/v1/duration.go#L487)? I believe there are some serialization issues with `*time.Duration`.